config/config.go

```diff
@@ -167,6 +166,21 @@ func setDefaults() {
 	viper.SetDefault(StatsFlushIntervalKey, DefaultStatsFlushInterval)
 }
 
+type Configurator interface {
+	SetDefault(key string, value interface{})
+}
+
+type viperConfigurator struct {
+}
+
+func (v viperConfigurator) SetDefault(key string, value interface{}) {
+	viper.SetDefault(key, value)
+}
+
+func (c *Config) Override(fn func(Configurator)) {
```

Comment:
Not a fan.  The viper configuration object seems global.  Certainly the call inside `viperConfigurator.SetDefault` is a global mutator.  This facility lets us hide a global mutation behind 3 layers of abstraction; I think it makes code that uses viper less safe.
I would rather just give a global `config.SetDefault` call by saying e.g.
```go
var SetDefault = viper.SetDefault
```

Reply:
it was a step towards instantiating a viper instance instead of using a global one. in general, current `config.NewConfig()` isn't really creating anything new, as config a wrapper around a global. I don't mind going with your direction and basically removing the config struct. All its methods can be functions anyway (but I don't want to tack that onto this PR).